### PR TITLE
8012: Add Excel Export functionality to DynamicForms

### LINKED ISSUE #8012: Add Excel Export functionality to DynamicForms
Dynamic Forms lacks a submissions export feature.

I achieved it locally with OpenXml and it exports submissions in an xlsx spreadsheet.
[here is the diff](https://github.com/Codinlab/Orchard/commit/fbe7f75d744d5241925ae656fd19cd0e9d18c346)

Is the community interested by a PR for that ?

### PR DESCRIPTION
**Pull request by TFleury**
closes #8012

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.DynamicForms/Services/IFormService.cs [hunk 1]
```diff
@@ -20,6 +21,7 @@ public interface IFormService : IDependency {
         Submission CreateSubmission(Submission submission);
         Submission GetSubmission(int id);
         IPageOfItems<Submission> GetSubmissions(string formName = null, int? skip = null, int? take = null);
+        Stream ExportSubmissions(string formName = null);
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.DynamicForms/Services/IFormService.cs:24]: Breaking change. Please create a new interface `IFormExportService` and have an `ExportXml` in it. Or target `dev`.

Comment (BenedekFarkas) [reply src/Orchard.Web/Modules/Orchard.DynamicForms/Services/IFormService.cs:24]: Does this matter still? I understand the concept, but an other application implementing this service is extremely unlikely.

**Comment by BenedekFarkas** [comment]
I see that `Maintainers are allowed to edit this pull request.`, but I can't push to the PR's branch, so I might create another PR from our fork with the same code. Anyway, I resolved the merge conflict and made small adjustments, otherwise this works great!

**Comment by sebastienros** [comment]
@TFleury can you add @BenedekFarkas as a contributor to https://github.com/Codinlab/Orchard/tree/DynamicFormsExport or make this repository accept contributions for PRs? He'd like to fix and merge this PR. Thanks.

**Comment by TFleury** [comment]
Done 👍

File: src/Orchard.Web/Modules/Orchard.DynamicForms/Services/IFormService.cs (hunk 1 above)
Comment (BenedekFarkas) [reply src/Orchard.Web/Modules/Orchard.DynamicForms/Services/IFormService.cs:24]: As discussed on the meeting, we can break stuff now. :D

**Comment by BenedekFarkas** [comment]
Thanks @TFleury, I've just pushed my changes!

**Review by BenedekFarkas — APPROVED** [review]

**Pull request merged**